ntbs-service/Pages/Notifications/NotificationEditModelBase.cs

```diff
@@ -26,6 +27,29 @@ public NotificationEditModelBase(INotificationService service, IAuthorizationSer
         [ViewData]
         public Dictionary<string, NotifyError> NotifyErrorDictionary { get; set; }
 
+        protected async Task SetNotificationAndAuthorize(int notificationId)
+        {
+            if (HttpContext.Request.Path.Value.Contains("ClinicalDetails"))
```

Comment:
Perhaps an abstract `GetNotification` (or `GetEntity`) method that children can provide?
Or a default one they can override?
Conditioning on the path here seems a little messy

Reply:
Good idea!